### PR TITLE
#15860: Resume action packets after unblocking (release PR)

### PR DESCRIPTION
**Pull request by polmr**
- unblock sc receiving after whyamiblocked returns unblock
- add some logging lines

### CODE REVIEW

File: src/megaclient.cpp [hunk 1]
```diff
@@ -2082,6 +2083,20 @@ void MegaClient::exec()
         }
         syncactivity = false;
 
+        if (stopsc || scpaused || !statecurrent || !syncsup)
+        {
+            LOG_verbose << " Megaclient exec is pending resolutions."
+                        << " scpaused=" << scpaused
+                        << " stopsc=" << stopsc
```
Comment (sergiohs84) [inline src/megaclient.cpp:2090]:
```suggestion
                        << " stopsc=" << stopsc
                        << " mScStoppedDueToBlock=" << mScStoppedDueToBlock
```

File: src/sync.cpp [hunk 2]
```diff
@@ -1301,7 +1301,7 @@ LocalNode* Sync::checkpath(LocalNode* l, string* localpath, string* localname, d
         isroot = l == localroot.get() && !newname.size();
     }
 
-    LOG_verbose << "Scanning: " << path;
+    LOG_verbose << "Scanning: " << path << " in=" << initializing << " full=" << fullscan << " l=" << l;
```
Comment (sergiohs84) [inline src/sync.cpp:1304]: `if (localname == nullptr) then l == nullptr`. Is it safe to pass `l` to the logger in that case?

File: include/mega/megaclient.h [hunk 3]
```diff
@@ -317,6 +317,12 @@ class MEGA_API MegaClient
     // check the reason of being blocked
     void whyamiblocked();
 
+    // block. This stops quering for action packets
```
Comment (sergiohs84) [inline include/mega/megaclient.h:320]:
```suggestion
    // stops querying for action packets due to the account being blocked (-16 on sc channel)
```

File: include/mega/megaclient.h [hunk 4]
```diff
@@ -317,6 +317,12 @@ class MEGA_API MegaClient
...
+    void block();
+
+    // called when unblocked. Resumes querying for action packets
```
Comment (sergiohs84) [inline include/mega/megaclient.h:323]:
```suggestion
    // resumes querying for action packets when the account is unblocked (if it was previously stopped due to being blocked)
```

File: src/sync.cpp (hunk 2 above)
Comment (polmr) [reply src/sync.cpp:1304]: yes, it'll print `NULL` or something similar

**Pull request merged**